--- a/src/bind.c	2017-02-03 16:16:55.346409392 -0800
+++ b/src/bind.c	2017-03-26 15:41:53.213261675 -0700
@@ -124,6 +124,16 @@
         CLOSE_SOCKET(s);
         return -1;
     }
+    if (!sshbind->blocking) {
+        if (ssh_socket_set_nonblocking(s) != 0) {
+            ssh_set_error(sshbind,
+                          SSH_FATAL,
+                          "Setting socket non-blocking failed: %s",
+                          strerror(errno));
+            CLOSE_SOCKET(s);
+            return -1;
+        }
+    }
 
     freeaddrinfo (ai);
     return s;
@@ -501,6 +511,18 @@
   return rc;
 }
 
+int ssh_event_add_bind(ssh_event event, ssh_bind sshbind) {
+    ssh_poll_handle h = ssh_bind_get_poll(sshbind);
+    if (!h) return SSH_ERROR;
+    return ssh_event_add_poll(event, h);
+}
+
+int ssh_event_remove_bind(ssh_event event, ssh_bind sshbind) {
+    ssh_poll_handle h = ssh_bind_get_poll(sshbind);
+    if (!h) return SSH_ERROR;
+    ssh_event_remove_poll(event, h);
+    return SSH_OK;
+}
 
 /**
  * @}
